### PR TITLE
mouse.c: fix mouse drag positions on multigrid

### PR DESCRIPTION
**Pull request by vhakulinen**
Currently, multigrid mouse drag positions are handled incorrectly if the
drag event is not in the top left grid. Fix this by not adjusting the
position of the event in jump_to_mouse.

### CODE REVIEW

**Comment by norcalli** [comment]
A little more explanation of "handled incorrectly" would be good. It's not exactly clear why this is the right fix yet (for the commit comment and also me)

**Comment by vhakulinen** [comment]
From what I answered on gitter:
> the problem (how I observed it) is basically this: when sending mouse drag events with nvim_input_mouse, the col and row values are not read as zero based values for the current grid. I didn't check if there are workarounds, such as sending the events to the base grid or with a offset (offset being the window position).

From what I've understood, goneovim uses the old `nvim_input` events and fvim sends the mouse input events only to grid 1 with the new `nvim_input_mouse` (maybe @akiyosi and @yatli can confirm both of these cases). Meanwhile, gnvim sends mouse events to each grid with `nvim_input_mouse`.

**Comment by Kethku** [comment]
I'm running into this issue as well with neovide. The above solutions in goneovim and fvim don't handle floating windows properly, so clicking on will focus the underlying grid rather than the floating window which is not good.

**Comment by akiyosi** [comment]
@Kethku 
Thanks.
I noticed just that last night with test in external window.

**Comment by Kethku** [comment]
I believe we should be able to work around this issue for now by subtracting the window location from drag events, but that shouldn't be necessary and is a hack

**Comment by vhakulinen** [comment]
> I believe we should be able to work around this issue for now by subtracting the window location from drag events, but that shouldn't be necessary and is a hack

I dont think that would work for externalized windows.

**Comment by yatli** [comment]
@vhakulinen yes, can confirm with fvim. Currently it relies on neovim to magically translate the grid 1 mouse input to the correct grid, but it wouldn't work for externalized grids that's not anchored to `#1`

**Comment by Kethku** [comment]
@yatli I believe such an approach also doesn't work with floating windows

File: test/functional/ui/multigrid_spec.lua [hunk 1]
```diff
@@ -1963,6 +1964,66 @@ describe('ext_multigrid', function()
...
+  it('supports mouse drag with mouse=a', function()
+    command('set mouse=a')
+    command('vsplit')
+    command('wincmd l')
+    command('split')
+    command('enew')
+    feed('ifoo\nbar<esc>')
+
+    meths.input_mouse('left', 'press', '', 5, 0, 0)
+    wait()
```
Comment (justinmk) [inline test/functional/ui/multigrid_spec.lua:1976]: this was renamed to `poke_eventloop`

**Comment by justinmk** [comment]
@vhakulinen would you mind rebasing this. related PR was just merged: https://github.com/neovim/neovim/pull/15091

**Comment by vhakulinen** [comment]
Rebased.

**Comment by zeertzjq** [comment]
This seems to introduce a bug (#15659).

**Comment by numbcoder** [comment]
ping @vhakulinen

**Pull request merged**